packages/experience-builder-sdk/src/hooks/useFetchBySlug.spec.ts

```diff
@@ -0,0 +1,203 @@
+import { useFetchBySlug, useFetchBySlugArgs } from './useFetchBySlug';
+import { EntityStore } from '@contentful/experience-builder-core';
+import { renderHook, waitFor } from '@testing-library/react';
+import { compositionEntry } from '../../test/__fixtures__/composition';
+import { entries, assets } from '../../test/__fixtures__/entities';
+import type { ContentfulClientApi, Entry } from 'contentful';
+import type { ExternalSDKMode } from '@contentful/experience-builder-core/types';
+
+const experienceTypeId = 'layout';
+const localeCode = 'en-US';
+const slug = 'hello-world';
+
+let clientMock: ContentfulClientApi<undefined>;
+
+describe('useFetchBySlug', () => {
+  beforeEach(() => {
+    clientMock = {
+      getEntries: jest.fn().mockImplementation((data) => {
+        if ('sys.id[in]' in data) {
+          return Promise.resolve({ items: entries });
+        }
+
+        return Promise.resolve({ items: [compositionEntry] });
+      }),
+      getAssets: jest.fn().mockResolvedValue({ items: assets }),
+    } as unknown as ContentfulClientApi<undefined>;
+  });
+
+  it('should be defined', async () => {
+    const { result } = renderHook(useFetchBySlug, {
+      initialProps: {
+        client: clientMock,
+        mode: 'preview' as ExternalSDKMode,
+        slug,
+        experienceTypeId,
+        localeCode,
+      },
+    });
+
+    await waitFor(() => {
+      expect(result.current).toBeDefined();
+      expect(result.current.isLoading).toBe(true);
+    });
```

Comment:
[nitpick]
It's not breaking the test but more in general, I want to call out that `waitFor` should only wait for one assertion at a time. You can read more about the reasoning behind this [here](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#having-multiple-assertions-in-a-single-waitfor-callback).

```suggestion
    // Just wait until we reached a specific state after async things happened
    await waitFor(() => expect(result.current).toBeDefined());
    // Then run all the sync assertions as it is not expected that something else async will change the state again
    expect(result.current.isLoading).toBe(true);
```